src/odemis/acq/stitching/_registrar.py

```diff
@@ -427,3 +437,313 @@ def _compute_registration(self, tile, row, col):
+                    self.shifts['ver'][i].append(None)
+        else:
+            raise ValueError("Cannot insert multiple tiles at the same position.")
+
+        self.tiles[row][col] = tile
+        self.acq_order.append([row, col])
+        return row, col
+
+    def _get_shift(self, prev_tile, tile):
+        """
+        Calculates the shift  between the two tiles. It also returns an error metric based on the
+        normalized cross-correlation (ncc).
+
+        :param prev_tile: (DataArray) static tile to which other tile is compared
+        :param tile: (numpy.array) shifted tile
+        :returns: ((int, int), float) x shift, y shift, error (1 - cross-correlation value)
+        """
+        px_size = tile.metadata[model.MD_PIXEL_SIZE]
+        # The y-axis of the reference coordinate system used here is inverted compared to
+        # the metadata position.
+        exp_shift = ((tile.metadata[model.MD_POS][0] - prev_tile.metadata[model.MD_POS][0]) / px_size[0],
+                     (-tile.metadata[model.MD_POS][1] + prev_tile.metadata[model.MD_POS][1]) / px_size[1])
+
+        # Get the region of interest
+        if (abs(exp_shift[0]) >= tile.shape[1]) or (abs(exp_shift[1]) >= tile.shape[0]):
+            raise ValueError("There is no overlap between tiles for the shift given %s" % exp_shift)
+        if exp_shift[0] < 0:
+            l1, r1 = 0, tile.shape[1] + int(exp_shift[0])
+            l2, r2 = -int(exp_shift[0]), tile.shape[1]
+        else:
+            l1, r1 = int(exp_shift[0]), tile.shape[1]
+            l2, r2 = 0, tile.shape[1] - int(exp_shift[0])
+
+        if exp_shift[1] < 0:
+            t1, b1 = 0, tile.shape[0] + int(exp_shift[1])
+            t2, b2 = -int(exp_shift[1]), tile.shape[0]
+        else:
+            t1, b1 = int(exp_shift[1]), tile.shape[0]
+            t2, b2 = 0, tile.shape[0] - int(exp_shift[1])
+        prev_tile_roi = numpy.array(prev_tile)[t1:b1, l1:r1]
+        tile_roi = numpy.array(tile)[t2:b2, l2:r2]
+
+        # Calculate the shift
+        shift = MeasureShift(tile_roi, prev_tile_roi)
+        shift_total = numpy.subtract(exp_shift, shift)
+
+        # Measure accuracy (ncc value)
+        avg = (numpy.sum(prev_tile_roi) / prev_tile_roi.size,
+               numpy.sum(tile_roi) / tile_roi.size)
+        dist = (numpy.subtract(prev_tile_roi, avg[0]),
+                numpy.subtract(tile_roi, avg[1]))
+        covar = numpy.sum(dist[0] * dist[1]) / prev_tile_roi.size
+        var = (numpy.sum(numpy.power(dist[0], 2)) / prev_tile_roi.size,
+               numpy.sum(numpy.power(dist[1], 2)) / tile_roi.size)
+        stDev = (math.sqrt(var[0]), math.sqrt(var[1]))
+        if stDev[0] == 0 or stDev[1] == 0:
+            return exp_shift, 100
+        # Normalized cross correlation (values between -1 and 1)
+        ncc = (covar / (stDev[0] * stDev[1]))
+        # Later, we want to build a minimum spanning tree based on the ncc value. This however
```

Comment:
I think it could be clearer if the function didn't care about what the error value will be used for. Maybe just report something between 0->1. The caller is in charge of converting this "error" value into something that's handy for it. 
=> Move this conversion to after, where you pass the error values to the minimum spanning tree.